### ISSUE TITLE
Comportamiento extraño

### ISSUE THREAD

**Issue by jpavonabian** [metalalchemist/VeTube#13]
Vengo con un comportamiento algo extraño. Cuando ejecuto el programa, bien desde el .exe o bien desde la consola ejecutando el .py, el programa abre sin problema, pero silencia todo el ordenador y me obliga a reiniciar.
Tristemente no puedo aportar más detalles, pues no tengo forma de saber qué ocurre cuando se silencia.

**Comment by metalalchemist** [comment]
Sonará un  tanto tonto pero lo abres desde tu pr? O de la v2.7 o del commit que hice ayer

**Comment by jpavonabian** [comment]
No, no es tonto. Es de lógica pura. He probado desde los 3 sitios con idéntico resultado.

 

De: cesar verastegui ***@***.***> 
Enviado el: domingo, 3 de septiembre de 2023 17:55
Para: metalalchemist/VeTube ***@***.***>
CC: Jesús Pavón Abián ***@***.***>; Author ***@***.***>
Asunto: Re: [metalalchemist/VeTube] Comportamiento extraño (Issue #13)

 

Sonará un tanto tonto pero lo abres desde tu pr? O de la v2.7 o del commit que hice ayer

**Comment by metalalchemist** [comment]
Es como si se congelará un hilo pero al abrir el programa solo hay un hilo.
Salvó que el actualizador no avance.
Puedes abrir config y establecer updates en false? Cuéntame cómo te va

**Comment by jpavonabian** [comment]
Probado. Crashea.

Lo curioso es que en el otro ordenador con máquina virtual no pasa.

**Comment by metalalchemist** [comment]
Y este no es virtualizado? Y es 64 o 32bits

**Comment by jpavonabian** [comment]
No. Es de 64 bits, un Windows 11.

Por lo que estoy viendo no parece ser del actualizador. Hay sonido hasta que el programa carga. Una vez ha cargado y está la interfaz visible, adiós al sonido del ordenador por completo hasta que reinicio, que supongo que se cargará algún proceso específico.

 

 

 

De: cesar verastegui ***@***.***> 
Enviado el: domingo, 3 de septiembre de 2023 18:37
Para: metalalchemist/VeTube ***@***.***>
CC: Jesús Pavón Abián ***@***.***>; Author ***@***.***>
Asunto: Re: [metalalchemist/VeTube] Comportamiento extraño (Issue #13)

 

Y este no es virtualizado? Y es 64 o 32bits

**Comment by metalalchemist** [comment]
Entonces no te sale el diálogo de donaciones si quiera?

**Comment by jpavonabian** [comment]
Sí, la interfaz carga perfectamente y el ordenador se puede usar, pero sin sonido.

 

 

De: cesar verastegui ***@***.***> 
Enviado el: domingo, 3 de septiembre de 2023 20:06
Para: metalalchemist/VeTube ***@***.***>
CC: Jesús Pavón Abián ***@***.***>; Author ***@***.***>
Asunto: Re: [metalalchemist/VeTube] Comportamiento extraño (Issue #13)

 

Entonces no te sale el diálogo de donaciones si quiera?

**Comment by metalalchemist** [comment]
Osea no suena ni el narrador

Prueba quitando la importación de playsound?

**Comment by jpavonabian** [comment]
El problema es que quitar playsound no debería ser motivo de fallo.

He modificado el json y he puesto toda cosa de sonido a false, y sigue explotando y tal como está montado, no debería reproducir nada, por lo que deberíamos poder descartarlo.

¿Hay algún sitio donde se silencie el ordenador o algo así?

Y lo más importante, por qué con un reinicio se soluciona? Hablo de reiniciar con shuutdown -f -r -t 0

¿el programa cuenta con alguna forma de levantarlo en modo debug para ver qué hace antes de dejarme sin sonido?

 

 

De: cesar verastegui ***@***.***> 
Enviado el: domingo, 3 de septiembre de 2023 20:17
Para: metalalchemist/VeTube ***@***.***>
CC: Jesús Pavón Abián ***@***.***>; Author ***@***.***>
Asunto: Re: [metalalchemist/VeTube] Comportamiento extraño (Issue #13)

 

Osea no suena ni el narrador

Prueba quitando la importación de playsound?

**Comment by metalalchemist** [comment]
Y si haces una conexión con nvdaremote al PC con otro PC para leer que arroja?
No veo creado nada debug

**Comment by jpavonabian** [comment]
Pues no arroja nada. Habría que depurarlo a mano para ver dónde se carga el sonido y por qué.

 

 

De: cesar verastegui ***@***.***> 
Enviado el: domingo, 3 de septiembre de 2023 20:48
Para: metalalchemist/VeTube ***@***.***>
CC: Jesús Pavón Abián ***@***.***>; Author ***@***.***>
Asunto: Re: [metalalchemist/VeTube] Comportamiento extraño (Issue #13)

 

Y si haces una conexión con nvdaremote al PC con otro PC para leer que arroja?
No veo creado nada debug

**Comment by rmcpantoja** [comment]
Hola @jpavonabian,
Otra librería que se usa aquí es sounddevice para la reproducción de audio en el sintetizador Piper, así que es provable que esté causando algún tipo de interferencia.
¿Puedes quitar la importación [de este archivo](https://github.com/metalalchemist/VeTube/blob/master/TTS/Piper/speaker.py#4) o cambiar a la rama de experimentación piper-soundlib para verificar si hay el mismo comportamiento?

**Comment by jpavonabian** [comment]
Pues tengo noticias de esto.

 

He ejecutado el programa tirando de NVDARemote y funcionar funciona perfecto.

Probaré esta tarde a pelearme con sounddevice.

 

 

De: Mateo Cedillo ***@***.***> 
Enviado el: lunes, 4 de septiembre de 2023 3:32
Para: metalalchemist/VeTube ***@***.***>
CC: Jesús Pavón Abián ***@***.***>; Mention ***@***.***>
Asunto: Re: [metalalchemist/VeTube] Comportamiento extraño (Issue #13)

 

Hola @jpavonabian <https://github.com/jpavonabian> ,
Otra librería que se usa aquí es sounddevice para la reproducción de audio en el sintetizador Piper, así que es provable que esté causando algún tipo de interferencia.
¿Puedes quitar la importación de este archivo <https://github.com/metalalchemist/VeTube/blob/master/TTS/Piper/speaker.py#4>  o cambiar a la rama de experimentación piper-soundlib para verificar si hay el mismo comportamiento?

**Comment by jpavonabian** [comment]
Resuelto el misterio.

Era cosa de actualizar el driver de audio.

Mil perdones por las molestias.

 

 

De: ***@***.*** ***@***.***> 
Enviado el: lunes, 4 de septiembre de 2023 9:25
Para: 'metalalchemist/VeTube' ***@***.***>
Asunto: RE: [metalalchemist/VeTube] Comportamiento extraño (Issue #13)

 

Pues tengo noticias de esto.

 

He ejecutado el programa tirando de NVDARemote y funcionar funciona perfecto.

Probaré esta tarde a pelearme con sounddevice.

 

 

De: Mateo Cedillo ***@***.*** ***@***.***> > 
Enviado el: lunes, 4 de septiembre de 2023 3:32
Para: metalalchemist/VeTube ***@***.*** ***@***.***> >
CC: Jesús Pavón Abián ***@***.*** ***@***.***> >; Mention ***@***.*** ***@***.***> >
Asunto: Re: [metalalchemist/VeTube] Comportamiento extraño (Issue #13)

 

Hola @jpavonabian <https://github.com/jpavonabian> ,
Otra librería que se usa aquí es sounddevice para la reproducción de audio en el sintetizador Piper, así que es provable que esté causando algún tipo de interferencia.
¿Puedes quitar la importación de este archivo <https://github.com/metalalchemist/VeTube/blob/master/TTS/Piper/speaker.py#4>  o cambiar a la rama de experimentación piper-soundlib para verificar si hay el mismo comportamiento?

**Comment by jpavonabian** [comment]
Cierro el Isue, pues está resuelto.
